src/quantem/imaging/time_series.py

```diff
@@ -0,0 +1,255 @@
+
+        Parameters
+        ----------
+        im_ref: [NDArray, Any]
+            First image to use as reference.
+        im: [NDArray, Any]
+            Second image to shift with respect to the reference.
+        return_aligned_image = True
+            If True, returns the shifted second image.
+        window_size: [int, None] = 7
+            Size of the window around the subpixel position. If None, defaults to 7.
+
+        Returns
+        -------
+        shift: NDArray
+            shifted coordinates (x,y) between two images
+        im_shift: NDArray
+            shifted image with respect to the shifted coordinates
+        """
+        # Take 2D DFT of an image (im) and the image before it as reference (im_ref)
+        G_ref = np.fft.fft2(im_ref) 
+        G_a0 = np.fft.fft2(im) 
+        
+        im_corr = np.real(np.fft.ifft2(G_ref * np.conj(G_a0)))
+
+       # Index position of the maximum  
+        nrows, ncols = im_corr.shape
+        peak_r, peak_c = np.unravel_index(np.argmax(im_corr), im_corr.shape)
+        
+        # Constants
+        half = window_size // 2
+
+        # Build periodic window
+        # Padding of the extra pixels
+        r_idx = np.mod(np.arange(peak_r - half, peak_r + half + 1), nrows)
+        c_idx = np.mod(np.arange(peak_c - half, peak_c + half + 1), ncols)
+        window = im_corr[np.ix_(r_idx, c_idx)]
+
+        # Center of mass within window
+        com_r, com_c = ndimage.center_of_mass(window)
+
+        # Adjust back to full image coordinates (with periodicity)
+        shift_row = (((peak_r - half + com_r) + nrows/2) % nrows) - nrows/2
+        shift_col = (((peak_c - half + com_c) + ncols/2) % ncols) - ncols/2
+        shift = np.array((shift_row,shift_col))
+
+        if return_aligned_image:
+            kx_shift = -2*np.pi*np.fft.fftfreq(im.shape[0])
+            ky_shift = -2*np.pi*np.fft.fftfreq(im.shape[1])
+            im_shift = np.real(np.fft.ifft2(
+                G_a0 * np.exp(1j * (kx_shift[:,None] * shift[0] + ky_shift[None,:] * shift[1]))
+            ))
+            return shift,im_shift
+
+        else:
+            return shift
+        
+
+    def align_stack(
+        stack_pad: NDArray | Any, 
+        window_size: int | None = 7,
+        running_average_frames: float | int = 20.0,
+    ) -> NDArray:
+        """
+        Use 2D DFT cross correlation alignment on an entire image stack.
+
+        Parameters
+        ----------
+        stack_pad: [NDArray, Any]
+            Unaligned image stack data (time, x, y).
+        window_size: [int, None] = 7
+            Size of the window around the subpixel position. If None, defaults to 7.
+        running_average_frames: [float, int] = 20.0
+            Maximum number of images for the running average applied to the reference. If None, default is 20  .
+        
+        Returns
+        -------
+        stack_aligned: NDArray
+            Aligned image stack data (time, x, y).
+        dxy: NDArray
```

Comment:
typically in Python, what we return needs to be more somewhat English-langauge like. `stack_aligned` is quite interpretable from the two words alone but `dyx`, I wonder what this is because the function is called `align_stack` so as an end user, I might wonder... what's "dyx" 

is it the average of y and x? of course I can read the docstring but we want our API to be seamless so that scientists can use it.